cmd/utils.go

```diff
@@ -185,8 +190,32 @@ func migrateData(ctx context.Context, targetProfile profiles.TargetProfile, sour
 		}
 		fmt.Printf("Schema validated successfully for data migration for db %s\n", dbURI)
 	}
+
+	// If migration type is Minimal Downtime, validate if required resources can be generated
+	if !conv.UI && sourceProfile.Driver == constants.MYSQL && sourceProfile.Ty == profiles.SourceProfileTypeConfig && sourceProfile.Config.ConfigType == constants.DATAFLOW_MIGRATION {
+		spClient, err:= spinstanceadmin.NewInstanceAdminClientImpl(ctx)
+		if err != nil {
+			return nil, err
+		}
+		dsClient, err := datastreamclient.NewDatastreamClientImpl(ctx)
+		if err != nil {
+			return nil, err
+		}
+		storageclient, err := storageclient.NewStorageClientImpl(ctx)
+		if err != nil {
+			return nil, err
+		}
+		validateResource := conversion.NewValidateResourcesImpl(&spanneraccessor.SpannerAccessorImpl{}, spClient, &datastream_accessor.DatastreamAccessorImpl{},
+			dsClient, &storageaccessor.StorageAccessorImpl{}, storageclient)
+		err = validateResource.ValidateResourceGeneration(ctx, targetProfile.Conn.Sp.Project, targetProfile.Conn.Sp.Instance, sourceProfile, conv)
+		if err != nil {
```

Comment:
This code piece seem duplicated thrice, maybe can be moved to a function and called instead?

Reply:
The error handling is different in each case and so are the arguments, it will end up being the same number of LOC even if if I do it in a function

Reply:
What I meant was something like `func foo() error {...}`. 
and the callers will look something like:
```
err := foo()
if err != nil {
    return nil, err // or subcommands.ExitFailure() for other implementations.
}
```